mp/src/game/server/momentum/mom_triggers.cpp

```diff
@@ -1573,7 +1579,21 @@ bool CNoGrenadesZone::IsInsideNoGrenadesZone(CBaseEntity *pOther)
             if (!pSticky)
                 return true; // This is a rocket in a nonade zone and should fizzle
 
-            return !pNoGrenadeZone->m_bAirborneOnly || !pSticky->DidHitWorld();
+            switch (pNoGrenadeZone->m_iStickyFizzleType)
+            {
+                case 0:
+	                return pPlayer->m_nButtons & IN_ATTACK2 && pSticky->IsArmed() && (!pNoGrenadeZone->m_bAirborneOnly || !pSticky->DidHitWorld()); // player must det for fizzle
```

Comment:
Since we only call this on collision with world now we don't need the player's buttons anymore. Can do the logic with just `m_bAirborneOnly`, `DidHitWorld()` and `m_iStickyFizzleType`.